src/DiffResults.jl

```diff
@@ -36,7 +36,7 @@ Return `r::DiffResult`, with output value storage provided by `value` and output
 storage provided by `derivs`.
 
 In reality, `DiffResult` is an abstract supertype of two concrete types, `MutableDiffResult`
-and `ImmutableDiffResult`. If all `value`/`derivs` are all `Number`s or `StaticArray`s,
+and `ImmutableDiffResult`. If all `value`/`derivs` are all `Number`s or `StaticArrays.SArray`s,
```

Comment:
As discussed in #18, it should rather be
```suggestion
and `ImmutableDiffResult`. If `value` or any element of `derivs` is a `Number` or `StaticArrays.SArray`,
```